politeiad/config.go

```diff
@@ -129,11 +137,42 @@ func supportedSubsystems() []string {
 	return subsystems
 }
 
+func parseAndValidateMimeTypes(mimeTypes string) ([]string, error) {
+	mimes := strings.Split(mimeTypes, ",")
+	parsed := make([]string, 0, len(mimes))
+
+	for _, m := range mimes {
+		mediatype, params, err := mime.ParseMediaType(m)
+		if err != nil {
+			return nil, err
+		}
+		// Check if the input media type is recognized by the mime
+		// package. This is a rough indication of whether the input
+		// media type is valid. The mime package does not include
+		// all media types so just throw a warning if the input
+		// media type is not recognized.
+		ext, err := mime.ExtensionsByType(mediatype)
+		if err != nil {
+			return nil, err
+		} else if len(ext) == 0 {
+			log.Warnf("MIME type '%v' not recognized", m)
```

Comment:
Shouldn't this be fatal?

Reply:
@lukebp and I discussed this and it was decided to just throw a warning due to the fact that the mime package in Go doesn't have all mime types included out of the box.

Reply:
If we have a chicken and egg problem here we probably have to remove the mime stuff from util and move it into d completely. Then when we do the policy call we return the supported mime types. I think this is actually what we need to do because this is looking very clunky.

Reply:
I'm not sure I understood the next steps correctly, marco. I think the mime stuff is basically all in d already; in `util/file.go` we use `politeiad/api/v1/mime` to do some mime checks on handling files. Is this what you mean that should go to d completely?

Reply:
Well, the code from util probably needs to live in www and use the policy mime types it received because the defaults no longer apply. This will break some things and those need to be fixed accordingly (like CLI tools).